data_source/gateway.go

```diff
@@ -0,0 +1,84 @@
+package dataSource
```

Comment:
Since this file is an interface between the gateway (and probably P2P) and core types, the file name should be something like `adapter.go`. This is not specific to the feeder gateway and the naming shouldn't have to change when we switch to say P2P in the future.

Reply:
P2P happens over JSON RPC, so that is a different protocol. If `adapt*` family of functions turn out to be reusable, we can extract them in to another package when it comes to that.